Makefile

```diff
@@ -49,10 +49,14 @@ else
 export GOTESTCOMMAND=gotestsum --format pkgname --jsonfile testresults.json --
 endif
 
-# M1 Mac--homebrew install location in /opt/homebrew
 ifeq ($(OS_TYPE), darwin)
-ifeq ($(ARCH), arm64)
+# For Xcode >= 15, set -no_warn_duplicate_libraries linker option
+XCODE_MAJOR_VERSION := $(shell xcodebuild -version | grep Xcode | awk '{print $$2}' | cut -d. -f1)
```

Comment:
Do we know if this works under CircleCI? I can't run `xcodebuild -version` on my local mac as I get this error:

```
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
```

Although this seems to work: `pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | grep version`

Reply:
@onetechnical maybe your local xcode is misconfigured? https://stackoverflow.com/a/17980786

Mine works with the original command